### PR TITLE
BDOG-886: 

### PR DESCRIPTION
**Pull request by JoPintoPaul**

### CODE REVIEW

File: project/AppDependencies.scala [hunk 1]
```diff
@@ -50,16 +50,47 @@ object AppDependencies {
...
   )
 
-  val testPlay27 = Seq(
+  val httpVerbsCoreTestPlay27 = Seq(
     "com.typesafe.play" %% "play-test"   % play27Version % Test
   )
+
+  val httpVerbsTestCompileCommon = Seq(
+    "org.scalatest"          %% "scalatest"      % "3.1.1",
+    "org.pegdown"            %  "pegdown"        % "1.6.0",
```
Comment (colin-lamed) [inline project/AppDependencies.scala:68]: can we remove pegdown?

File: project/AppDependencies.scala [hunk 2]
```diff
@@ -6,7 +6,7 @@ object AppDependencies {
   val play26Version = "2.6.20"
   val play27Version = "2.7.4"
 
-  val compileCommon = Seq(
+  val httpVerbsCoreCompileCommon = Seq(
```
Comment (colin-lamed) [inline project/AppDependencies.scala:9]: drop the `httpVerbs` prefix - it's a given?

File: http-verbs-common/src/test/scala/uk/gov/hmrc/http/HttpGetSpec.scala [hunk 3]
```diff
@@ -102,10 +101,30 @@ class HttpGetSpec
...
       testGet.GET[HttpResponse](url, Seq("header" -> "foo")).futureValue shouldBe response
     }
+
     "be able to return objects deserialised from JSON" in {
       val testGet = new StubbedHttpGet(Future.successful(HttpResponse(200, """{"foo":"t","bar":10}""")))
       testGet.GET[TestClass](url, Seq("header" -> "foo")).futureValue should be(TestClass("t", 10))
     }
+
+    "be able to return Some[T] when deserialising as an option of object from JSON" in {
+      val testGet = new StubbedHttpGet(Future.successful(HttpResponse(200, """{"foo":"t","bar":10}""")))
```
Comment (colin-lamed) [inline http-verbs-common/src/test/scala/uk/gov/hmrc/http/HttpGetSpec.scala:111]: Is this one of the tests from `http-verbs-example`? I think the style in the HeadersSpec serves as a better example of how to use the library - i.e. creating a client.
Maybe we need a separate class to demonstrate how to use the library, separate from us testing that the library behaves as expected. The intention of the tests will be clear when maintaining the library. And we can create a link from the README to tell users to look at those tests.

File: project/AppDependencies.scala (hunk 1 above)
Comment (JoPintoPaul) [reply project/AppDependencies.scala]: Good shout, done.

File: project/AppDependencies.scala (hunk 2 above)
Comment (JoPintoPaul) [reply project/AppDependencies.scala]: “You should name a variable using the same care with which you name a first-born child.”

Agreed, done.

File: http-verbs-common/src/test/scala/uk/gov/hmrc/http/HttpGetSpec.scala (hunk 3 above)
Comment (JoPintoPaul) [reply http-verbs-common/src/test/scala/uk/gov/hmrc/http/HttpGetSpec.scala:111]: Have added Examples as separate tests in the project and linked from the README.

**Pull request merged**